src/SAML2/XML/mdui/UIInfo.php

```diff
@@ -334,16 +334,24 @@ public static function fromXML(DOMElement $xml): object
         Assert::same($xml->localName, 'UIInfo', InvalidDOMElementException::class);
         Assert::same($xml->namespaceURI, UIInfo::NS, InvalidDOMElementException::class);
 
-        $DisplayName = XMLUtils::extractLocalizedStrings($xml, UIInfo::NS, 'DisplayName');
-        $Description = XMLUtils::extractLocalizedStrings($xml, UIInfo::NS, 'Description');
-        $InformationURL = XMLUtils::extractLocalizedStrings($xml, UIInfo::NS, 'InformationURL');
-        $PrivacyStatementURL = XMLUtils::extractLocalizedStrings($xml, UIInfo::NS, 'PrivacyStatementURL');
-        $Keywords = $Logo = $children = [];
+        $DisplayName = $Description = $InformationURL = $PrivacyStatementURL = $Keywords = $Logo = $children = [];
 
         /** @var \DOMElement $node */
         foreach (XMLUtils::xpQuery($xml, './*') as $node) {
             if ($node->namespaceURI === UIInfo::NS) {
                 switch ($node->localName) {
+                    case 'DisplayName':
+                        $DisplayName[] = DisplayName::fromXML($node);
+                        break;
+                    case 'Description':
+                        $Description[] = Description::fromXML($node);
+                        break;
+                    case 'InformationURL':
+                        $InformationURL[] = InformationURL::fromXML($node);
+                        break;
+                    case 'PrivacyStatementURL':
+                        $PrivacyStatementURL[] = PrivacyStatementURL::fromXML($node);
+                        break;
```

Comment:
Couldn't we here use `getChildrenOfClass()`?

Reply:
We could rewrite this entire method to:
- Use getChildrenOfClass for everything we need and;
- Use the for-loop to catch everything else

Ideally we'd want to avoid creating Chunks for classes we know, but that would require our class registry in place